File: admin_platform/android/app/src/main/java/com/tigerwallet/admin/data/api/ApiService.kt
package com.tigerwallet.admin.data.api

import com.tigerwallet.admin.data.api.model.*
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    // Auth
    @POST("api/v1/auth/login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @POST("api/v1/auth/logout")
    suspend fun logout(): Response<Unit>

    @POST("api/v1/auth/refresh")
    suspend fun refreshToken(@Body request: RefreshTokenRequest): Response<LoginResponse>

    @GET("api/v1/auth/me")
    suspend fun getCurrentAdmin(): Response<AdminResponse>

    // Users
    @GET("api/v1/users")
    suspend fun getUsers(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("kyc_status") kycStatus: String? = null,
        @Query("search") search: String? = null
    ): Response<UsersResponse>

    @GET("api/v1/users/{id}")
    suspend fun getUser(@Path("id") id: String): Response<UserResponse>

    @PUT("api/v1/users/{id}")
    suspend fun updateUser(@Path("id") id: String, @Body request: UpdateUserRequest): Response<UserResponse>

    @POST("api/v1/users/{id}/suspend")
    suspend fun suspendUser(@Path("id") id: String, @Body request: SuspendRequest): Response<Unit>

    @POST("api/v1/users/{id}/ban")
    suspend fun banUser(@Path("id") id: String, @Body request: BanRequest): Response<Unit>

    // KYC
    @GET("api/v1/kyc")
    suspend fun getKYCSubmissions(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("level") level: Int? = null
    ): Response<KYCListResponse>

    @GET("api/v1/kyc/{id}")
    suspend fun getKYC(@Path("id") id: String): Response<KYCResponse>

    @POST("api/v1/kyc/{id}/approve")
    suspend fun approveKYC(@Path("id") id: String, @Body request: ApproveKYCRequest): Response<Unit>

    @POST("api/v1/kyc/{id}/reject")
    suspend fun rejectKYC(@Path("id") id: String, @Body request: RejectKYCRequest): Response<Unit>

    // Tokens
    @GET("api/v1/tokens")
    suspend fun getTokens(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("chain") chain: String? = null,
        @Query("search") search: String? = null
    ): Response<TokensResponse>

    @GET("api/v1/tokens/{id}")
    suspend fun getToken(@Path("id") id: String): Response<TokenResponse>

    @POST("api/v1/tokens")
    suspend fun createToken(@Body request: CreateTokenRequest): Response<TokenResponse>

    @PUT("api/v1/tokens/{id}")
    suspend fun updateToken(@Path("id") id: String, @Body request: UpdateTokenRequest): Response<TokenResponse>

    @DELETE("api/v1/tokens/{id}")
    suspend fun deleteToken(@Path("id") id: String): Response<Unit>

    @POST("api/v1/tokens/{id}/verify")
    suspend fun verifyToken(@Path("id") id: String): Response<Unit>

    // Pairs
    @GET("api/v1/pairs")
    suspend fun getPairs(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("chain") chain: String? = null
    ): Response<PairsResponse>

    @GET("api/v1/pairs/{id}")
    suspend fun getPair(@Path("id") id: String): Response<PairResponse>

    @POST("api/v1/pairs")
    suspend fun createPair(@Body request: CreatePairRequest): Response<PairResponse>

    @PUT("api/v1/pairs/{id}")
    suspend fun updatePair(@Path("id") id: String, @Body request: UpdatePairRequest): Response<PairResponse>

    @DELETE("api/v1/pairs/{id}")
    suspend fun deletePair(@Path("id") id: String): Response<Unit>

    // Transactions
    @GET("api/v1/transactions")
    suspend fun getTransactions(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("type") type: String? = null,
        @Query("user_id") userId: String? = null
    ): Response<TransactionsResponse>

    @GET("api/v1/transactions/{id}")
    suspend fun getTransaction(@Path("id") id: String): Response<TransactionResponse>

    // Withdrawals
    @GET("api/v1/withdrawals")
    suspend fun getWithdrawals(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null
    ): Response<WithdrawalsResponse>

    @POST("api/v1/withdrawals/{id}/approve")
    suspend fun approveWithdrawal(@Path("id") id: String): Response<Unit>

    @POST("api/v1/withdrawals/{id}/reject")
    suspend fun rejectWithdrawal(@Path("id") id: String, @Body request: RejectWithdrawalRequest): Response<Unit>

    // Chains
    @GET("api/v1/chains")
    suspend fun getChains(): Response<ChainsResponse>

    @GET("api/v1/chains/{id}")
    suspend fun getChain(@Path("id") id: String): Response<ChainResponse>

    @POST("api/v1/chains")
    suspend fun createChain(@Body request: CreateChainRequest): Response<ChainResponse>

    @PUT("api/v1/chains/{id}")
    suspend fun updateChain(@Path("id") id: String, @Body request: UpdateChainRequest): Response<ChainResponse>

    // Fees
    @GET("api/v1/fees")
    suspend fun getFees(): Response<FeesResponse>

    @POST("api/v1/fees")
    suspend fun createFee(@Body request: CreateFeeRequest): Response<FeeResponse>

    @PUT("api/v1/fees/{id}")
    suspend fun updateFee(@Path("id") id: String, @Body request: UpdateFeeRequest): Response<FeeResponse>

    // White Labels
    @GET("api/v1/white-labels")
    suspend fun getWhiteLabels(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null
    ): Response<WhiteLabelsResponse>

    @GET("api/v1/white-labels/{id}")
    suspend fun getWhiteLabel(@Path("id") id: String): Response<WhiteLabelResponse>

    @POST("api/v1/white-labels")
    suspend fun createWhiteLabel(@Body request: CreateWhiteLabelRequest): Response<WhiteLabelResponse>

    @PUT("api/v1/white-labels/{id}")
    suspend fun updateWhiteLabel(@Path("id") id: String, @Body request: UpdateWhiteLabelRequest): Response<WhiteLabelResponse>

    @POST("api/v1/white-labels/{id}/approve")
    suspend fun approveWhiteLabel(@Path("id") id: String): Response<Unit>

    @POST("api/v1/white-labels/{id}/suspend")
    suspend fun suspendWhiteLabel(@Path("id") id: String, @Body request: SuspendWhiteLabelRequest): Response<Unit>

    // Dashboard
    @GET("api/v1/dashboard")
    suspend fun getDashboardStats(): Response<DashboardResponse>

    @GET("api/v1/dashboard/analytics")
    suspend fun getAnalytics(@Query("period") period: String = "24h"): Response<AnalyticsResponse>

    // Admins
    @GET("api/v1/admins")
    suspend fun getAdmins(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("role") role: String? = null
    ): Response<AdminsResponse>

    @POST("api/v1/admins")
    suspend fun createAdmin(@Body request: CreateAdminRequest): Response<AdminResponse>

    @PUT("api/v1/admins/{id}")
    suspend fun updateAdmin(@Path("id") id: String, @Body request: UpdateAdminRequest): Response<AdminResponse>

    @DELETE("api/v1/admins/{id}")
    suspend fun deleteAdmin(@Path("id") id: String): Response<Unit>

    // Audit Logs
    @GET("api/v1/audit")
    suspend fun getAuditLogs(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("admin_id") adminId: String? = null,
        @Query("action") action: String? = null
    ): Response<AuditLogsResponse>

    // Tickets
    @GET("api/v1/tickets")
    suspend fun getTickets(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("priority") priority: String? = null,
        @Query("category") category: String? = null
    ): Response<TicketsResponse>

    @GET("api/v1/tickets/{id}")
    suspend fun getTicket(@Path("id") id: String): Response<TicketResponse>

    @POST("api/v1/tickets")
    suspend fun createTicket(@Body request: CreateTicketRequest): Response<TicketResponse>

    @PUT("api/v1/tickets/{id}")
    suspend fun updateTicket(@Path("id") id: String, @Body request: UpdateTicketRequest): Response<TicketResponse>

    @POST("api/v1/tickets/{id}/messages")
    suspend fun addTicketMessage(@Path("id") id: String, @Body request: AddMessageRequest): Response<MessageResponse>

    // Knowledge Base
    @GET("api/v1/knowledge-base")
    suspend fun getKnowledgeBaseArticles(): Response<ArticlesResponse>

    @GET("api/v1/knowledge-base/{id}")
    suspend fun getKnowledgeBaseArticle(@Path("id") id: String): Response<ArticleResponse>

    @POST("api/v1/knowledge-base")
    suspend fun createKnowledgeBaseArticle(@Body request: CreateArticleRequest): Response<ArticleResponse>

    @PUT("api/v1/knowledge-base/{id}")
    suspend fun updateKnowledgeBaseArticle(@Path("id") id: String, @Body request: UpdateArticleRequest): Response<ArticleResponse>

    // Approval Workflows
    @GET("api/v1/workflows")
    suspend fun getWorkflows(): Response<WorkflowsResponse>

    @POST("api/v1/workflows")
    suspend fun createWorkflow(@Body request: CreateWorkflowRequest): Response<WorkflowResponse>

    @GET("api/v1/approval-requests")
    suspend fun getApprovalRequests(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null,
        @Query("workflow_id") workflowId: String? = null
    ): Response<ApprovalRequestsResponse>

    @POST("api/v1/approval-requests/{id}/approve")
    suspend fun approveRequest(@Path("id") id: String): Response<Unit>

    @POST("api/v1/approval-requests/{id}/reject")
    suspend fun rejectRequest(@Path("id") id: String, @Body request: RejectRequestRequest): Response<Unit>

    // Dashboards
    @GET("api/v1/dashboard/compliance")
    suspend fun getComplianceDashboard(): Response<ComplianceDashboardResponse>

    @GET("api/v1/dashboard/finance")
    suspend fun getFinanceDashboard(): Response<FinanceDashboardResponse>

    @GET("api/v1/dashboard/security")
    suspend fun getSecurityDashboard(): Response<SecurityDashboardResponse>

    // Notifications
    @GET("api/v1/notifications")
    suspend fun getNotifications(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null
    ): Response<NotificationsResponse>

    @PUT("api/v1/notifications/{id}/read")
    suspend fun markNotificationRead(@Path("id") id: String): Response<Unit>

    @PUT("api/v1/notifications/read-all")
    suspend fun markAllNotificationsRead(): Response<Unit>

    @POST("api/v1/notifications")
    suspend fun sendNotification(@Body request: SendNotificationRequest): Response<NotificationResponse>

    @POST("api/v1/notifications/broadcast")
    suspend fun broadcastNotification(@Body request: BroadcastNotificationRequest): Response<Unit>

    // API Keys
    @GET("api/v1/api-keys")
    suspend fun getAPIKeys(): Response<APIKeysResponse>

    @POST("api/v1/api-keys")
    suspend fun createAPIKey(@Body request: CreateAPIKeyRequest): Response<APIKeyResponse>

    @PUT("api/v1/api-keys/{id}")
    suspend fun updateAPIKey(@Path("id") id: String, @Body request: UpdateAPIKeyRequest): Response<APIKeyResponse>

    @POST("api/v1/api-keys/{id}/revoke")
    suspend fun revokeAPIKey(@Path("id") id: String): Response<Unit>

    // Webhooks
    @GET("api/v1/webhooks")
    suspend fun getWebhooks(): Response<WebhooksResponse>

    @POST("api/v1/webhooks")
    suspend fun createWebhook(@Body request: CreateWebhookRequest): Response<WebhookResponse>

    @PUT("api/v1/webhooks/{id}")
    suspend fun updateWebhook(@Path("id") id: String, @Body request: UpdateWebhookRequest): Response<WebhookResponse>

    @DELETE("api/v1/webhooks/{id}")
    suspend fun deleteWebhook(@Path("id") id: String): Response<Unit>

    // Market Maker Bots
    @GET("api/v1/market-maker")
    suspend fun getMarketMakerBots(
        @Query("page") page: Int = 1,
        @Query("limit") limit: Int = 20,
        @Query("status") status: String? = null
    ): Response<BotsResponse>

    @GET("api/v1/market-maker/{id}")
    suspend fun getMarketMakerBot(@Path("id") id: String): Response<BotResponse>

    @POST("api/v1/market-maker/{id}/start")
    suspend fun startBot(@Path("id") id: String): Response<Unit>

    @POST("api/v1/market-maker/{id}/stop")
    suspend fun stopBot(@Path("id") id: String): Response<Unit>

    @POST("api/v1/market-maker/{id}/pause")
    suspend fun pauseBot(@Path("id") id: String): Response<Unit>
}

File: admin_platform/flutter/lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ApiService {
  static const String baseUrl = 'https://admin-api.tigerwallet.com';
  static String? _token;

  static void setToken(String token) {
    _token = token;
  }

  static Map<String, String> get _headers => {
    'Content-Type': 'application/json',
    if (_token != null) 'Authorization': 'Bearer $_token',
  };

  // Auth
  static Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/auth/login'),
      headers: _headers,
      body: jsonEncode({'email': email, 'password': password}),
    );
    final data = jsonDecode(response.body);
    if (data['token'] != null) {
      setToken(data['token']);
    }
    return data;
  }

  static Future<void> logout() async {
    await http.post(Uri.parse('$baseUrl/api/v1/auth/logout'), headers: _headers);
    _token = null;
  }

  static Future<Map<String, dynamic>> getCurrentAdmin() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/auth/me'), headers: _headers);
    return jsonDecode(response.body);
  }

  // Users
  static Future<Map<String, dynamic>> getUsers({int page = 1, int limit = 20, String? status, String? search}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (search != null) queryParams['search'] = search;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/users').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> getUser(String id) async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/users/$id'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> updateUser(String id, Map<String, dynamic> data) async {
    final response = await http.put(
      Uri.parse('$baseUrl/api/v1/users/$id'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> suspendUser(String id, String reason) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/users/$id/suspend'),
      headers: _headers,
      body: jsonEncode({'reason': reason}),
    );
  }

  static Future<void> banUser(String id, String reason) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/users/$id/ban'),
      headers: _headers,
      body: jsonEncode({'reason': reason}),
    );
  }

  // KYC
  static Future<Map<String, dynamic>> getKYCSubmissions({int page = 1, int limit = 20, String? status, int? level}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (level != null) queryParams['level'] = level.toString();
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/kyc').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<void> approveKYC(String id, {String? notes}) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/kyc/$id/approve'),
      headers: _headers,
      body: jsonEncode(notes != null ? {'notes': notes} : {}),
    );
  }

  static Future<void> rejectKYC(String id, String reason) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/kyc/$id/reject'),
      headers: _headers,
      body: jsonEncode({'reason': reason}),
    );
  }

  // Tokens
  static Future<Map<String, dynamic>> getTokens({int page = 1, int limit = 20, String? status, String? chain}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (chain != null) queryParams['chain'] = chain;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/tokens').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createToken(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/tokens'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> updateToken(String id, Map<String, dynamic> data) async {
    final response = await http.put(
      Uri.parse('$baseUrl/api/v1/tokens/$id'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> deleteToken(String id) async {
    await http.delete(Uri.parse('$baseUrl/api/v1/tokens/$id'), headers: _headers);
  }

  // Pairs
  static Future<Map<String, dynamic>> getPairs({int page = 1, int limit = 20, String? status, String? chain}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (chain != null) queryParams['chain'] = chain;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/pairs').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createPair(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/pairs'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  // Transactions
  static Future<Map<String, dynamic>> getTransactions({int page = 1, int limit = 20, String? status, String? type}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (type != null) queryParams['type'] = type;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/transactions').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  // Withdrawals
  static Future<Map<String, dynamic>> getWithdrawals({int page = 1, int limit = 20, String? status}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/withdrawals').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<void> approveWithdrawal(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/withdrawals/$id/approve'), headers: _headers);
  }

  static Future<void> rejectWithdrawal(String id, String reason) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/withdrawals/$id/reject'),
      headers: _headers,
      body: jsonEncode({'reason': reason}),
    );
  }

  // Chains
  static Future<Map<String, dynamic>> getChains() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/chains'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createChain(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/chains'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  // Fees
  static Future<Map<String, dynamic>> getFees() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/fees'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createFee(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/fees'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  // White Labels
  static Future<Map<String, dynamic>> getWhiteLabels({int page = 1, int limit = 20, String? status}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/white-labels').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createWhiteLabel(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/white-labels'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  // Dashboard
  static Future<Map<String, dynamic>> getDashboard() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/dashboard'), headers: _headers);
    return jsonDecode(response.body);
  }

  // Tickets
  static Future<Map<String, dynamic>> getTickets({int page = 1, int limit = 20, String? status, String? priority, String? category}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (priority != null) queryParams['priority'] = priority;
    if (category != null) queryParams['category'] = category;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/tickets').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createTicket(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/tickets'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> updateTicket(String id, Map<String, dynamic> data) async {
    final response = await http.put(
      Uri.parse('$baseUrl/api/v1/tickets/$id'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> addTicketMessage(String ticketId, String message) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/tickets/$ticketId/messages'),
      headers: _headers,
      body: jsonEncode({'message': message}),
    );
  }

  // Knowledge Base
  static Future<Map<String, dynamic>> getKnowledgeBaseArticles() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/knowledge-base'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createKnowledgeBaseArticle(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/knowledge-base'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  // Approval Workflows
  static Future<Map<String, dynamic>> getWorkflows() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/workflows'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createWorkflow(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/workflows'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> getApprovalRequests({int page = 1, int limit = 20, String? status, String? workflowId}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    if (workflowId != null) queryParams['workflow_id'] = workflowId;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/approval-requests').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<void> approveRequest(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/approval-requests/$id/approve'), headers: _headers);
  }

  static Future<void> rejectRequest(String id, String reason) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/approval-requests/$id/reject'),
      headers: _headers,
      body: jsonEncode({'reason': reason}),
    );
  }

  // Dashboards
  static Future<Map<String, dynamic>> getComplianceDashboard() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/dashboard/compliance'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> getFinanceDashboard() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/dashboard/finance'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> getSecurityDashboard() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/dashboard/security'), headers: _headers);
    return jsonDecode(response.body);
  }

  // Notifications
  static Future<Map<String, dynamic>> getNotifications({int page = 1, int limit = 20, String? status}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/notifications').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<void> markNotificationRead(String id) async {
    await http.put(Uri.parse('$baseUrl/api/v1/notifications/$id/read'), headers: _headers);
  }

  static Future<void> sendNotification(Map<String, dynamic> data) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/notifications'),
      headers: _headers,
      body: jsonEncode(data),
    );
  }

  static Future<void> broadcastNotification(Map<String, dynamic> data) async {
    await http.post(
      Uri.parse('$baseUrl/api/v1/notifications/broadcast'),
      headers: _headers,
      body: jsonEncode(data),
    );
  }

  // API Keys
  static Future<Map<String, dynamic>> getAPIKeys() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/api-keys'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createAPIKey(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/api-keys'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> revokeAPIKey(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/api-keys/$id/revoke'), headers: _headers);
  }

  // Webhooks
  static Future<Map<String, dynamic>> getWebhooks() async {
    final response = await http.get(Uri.parse('$baseUrl/api/v1/webhooks'), headers: _headers);
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createWebhook(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/webhooks'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> updateWebhook(String id, Map<String, dynamic> data) async {
    final response = await http.put(
      Uri.parse('$baseUrl/api/v1/webhooks/$id'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> deleteWebhook(String id) async {
    await http.delete(Uri.parse('$baseUrl/api/v1/webhooks/$id'), headers: _headers);
  }

  // Market Maker Bots
  static Future<Map<String, dynamic>> getMarketMakerBots({int page = 1, int limit = 20, String? status}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (status != null) queryParams['status'] = status;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/market-maker').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<void> startBot(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/market-maker/$id/start'), headers: _headers);
  }

  static Future<void> stopBot(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/market-maker/$id/stop'), headers: _headers);
  }

  static Future<void> pauseBot(String id) async {
    await http.post(Uri.parse('$baseUrl/api/v1/market-maker/$id/pause'), headers: _headers);
  }

  // Admins (Super Admin)
  static Future<Map<String, dynamic>> getAdmins({int page = 1, int limit = 20, String? role}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (role != null) queryParams['role'] = role;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/admins').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> createAdmin(Map<String, dynamic> data) async {
    final response = await http.post(
      Uri.parse('$baseUrl/api/v1/admins'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<Map<String, dynamic>> updateAdmin(String id, Map<String, dynamic> data) async {
    final response = await http.put(
      Uri.parse('$baseUrl/api/v1/admins/$id'),
      headers: _headers,
      body: jsonEncode(data),
    );
    return jsonDecode(response.body);
  }

  static Future<void> deleteAdmin(String id) async {
    await http.delete(Uri.parse('$baseUrl/api/v1/admins/$id'), headers: _headers);
  }

  // Audit Logs
  static Future<Map<String, dynamic>> getAuditLogs({int page = 1, int limit = 20, String? adminId, String? action}) async {
    final queryParams = {'page': page.toString(), 'limit': limit.toString()};
    if (adminId != null) queryParams['admin_id'] = adminId;
    if (action != null) queryParams['action'] = action;
    
    final response = await http.get(
      Uri.parse('$baseUrl/api/v1/audit-logs').replace(queryParameters: queryParams),
      headers: _headers,
    );
    return jsonDecode(response.body);
  }
}
